nucliadb/nucliadb/ingest/orm/processor/__init__.py

```diff
@@ -677,3 +685,17 @@ async def delete_kb(self, kbid: str = "", slug: str = "") -> str:
             raise exc
         await txn.commit()
         return uuid
+
+
+def messages_source(messages: list[writer_pb2.BrokerMessage]):
+    from_writer = all(
+        (
+            message.source is writer_pb2.BrokerMessage.MessageSource.WRITER
+            for message in messages
+        )
+    )
+    if from_writer:
+        source = nodewriter_pb2.IndexMessageSource.WRITER
+    else:
+        source = nodewriter_pb2.IndexMessageSource.PROCESSOR
+    return source
```

Comment:
What happens if we we get messages from different sources in the same txn?

I know it may never be the case, but to be safer, maybe we can raise an error?

Reply:
Sure, I can raise an error. Anyway, it'd surprise me if we are using this feature haha

Reply:
I don't feel comfortable raising an exception. I'll capture one and log an error but default to PROCESSOR. If we don't support multi broker message transactions, we should remove all the code related.